File: checky-lua/checky.lua
#!/usr/local/bin/lua
-- /* """Lists how many images are in each brand's image processing folder""" */
-- /* __author__ = "Jason Rebuck" */
-- /* __copyright__ = "2021" */
-- /* __version__ = "0.10" */

require 'lfs'

-- find proofs folder
function get_base_path()
    for index,path in ipairs({'/Volumes/proofs', '/mnt/proofs'})
	do
	    if lfs.attributes(path:gsub("\\$",""),"mode") == "directory" 
	    then
		return path
	    end
	end
     kill_if_no_path()
end


-- exit if not proofs folder found
function kill_if_no_path()
    print("\027[1m\027[31mMissing Proofs Path!\027[0m\n")
    os.exit()
end

-- count the number of files in a folder
function count(path)
    if lfs.attributes(path:gsub("\\$",""),"mode") == "directory"
    then
	cnt = 0
	for file in lfs.dir(path)
	do
	    if string.match(file, "^[A-Za-z]")
	    then
		cnt = cnt + 1
	    end
	end
	return cnt
    else
	return -1
    end
end

-- print brand title
function print_title(brand)
    print(string.format("\n%s%11s%-9s%s", "\027[7m", brand:upper(), "", "\027[0m"))
end

-- print folder name, count and icon
function print_count(name, num)
    color = "\027[30m"
    icon = " "
    if num == 0
    then
        num = "▪"
    elseif num == -1
    then
        num = "-ERR-"
        color = "\027[2m"
    elseif num >= 50
    then
        color = "\027[5m\027[35m"
        icon = "☠️"
    elseif num >= 40
    then
        color = "\027[32m"
        icon = "🐲"
    elseif num >= 30
    then
        color = "\027[35m"
        icon = "🦄"
    elseif num >= 20
    then
        color = "\027[33m"
        icon = "🔥"
    else
        color = "\027[31m"
    end
    print(string.format(" \027[1m%s%-13s%s%s %s ", color, name:upper(), tostring(num), "\027[0m", icon))
end

-- loop through folders and print folder name and count
function print_status(brand, base, folders)
    print_title(brand)
    for index, folder in ipairs(folders)
    do
        print_count(folder, count(string.format("%s/images/bdp/%s/image_processing/%s", base, brand:lower(), folder)))
    end
end

-- main loop
function main()
    base_path = get_base_path()

    --HBD
    print_status("hbd", base_path, {"overlay", "portrait", "logo", "house", "pdf", "recipe", "postcard"})

    --YHL
    print_status("yhl", base_path, {"overlay", "portrait", "logo", "house", "pdf", "recipe", "postcard"})

    --RLP
    print_status("rlp", base_path, {"overlay", "portrait", "logo", "house", "pdf"})

    print("")
end

-- run main
main()

File: checky-js/checky.js
#!/usr/bin/node
const fs = require("fs")

// """Lists how many images are in each brand's image processing folder"""
// __author__ = "Jason Rebuck"
// __copyright__ = "2022"
// __version__ = "0.05"

//get base path
const get_base_path = () => {
    directory = "/Volumes/proofs"
    if (fs.existsSync(directory)) {
	return directory;
    } else {
	console.log("\033[1m\033[31mMissing Proofs Path!\033[0m\n");
	process.exit(1);
    }
}

//count files
const count = (path) => { 
    try {
	return fs.readdirSync(path).length;
    } catch(e) {
	return -1;
    }
};

//print section title
const print_title = (brand) => { console.log("\n\033[7m%s%s\33[0m", "".padStart(8), brand.toUpperCase().padEnd(10)) };

//print folder count
const print_count = (name, num) => { 
    icon = "";
    color = "\033[30m"
    if (num == 0) {  
	num = "▪"; 
    } else if (num == -1) {
	num = "-ERR-";
	color = "\033[30m\033[2m";
    } else if (num >= 50) {
	color = "\033[5m\033[35m";
	icon = "☠️";
    } else if (num >= 40) {
	color = "\033[32m";
	icon = "🐲";
    } else if (num >= 30) {
	color = "\033[35m";
	icon = "🦄";
    } else if (num >= 20) {
	color = "\033[33m";
	icon = "🔥";
    } else {
	color = "\033[31m";
	icon = "📌";
    }
    console.log(" \033[1m%s%s%s\033[0m %s", color, name.toUpperCase().padEnd(13), num.toString().padEnd(2), icon); 
};


//loop though folders
const print_status = (brand, base, folders) => {
    print_title(brand);
    folders.forEach(folder => { 
	print_count(folder, count(`${base}/images/bdp/${brand.toLowerCase()}/image_processing/${folder}`))
    });
}

// main
const main = () => {
    // find base path or exit
    let base = get_base_path();
    // hbd
    print_status("hbd", base, ["overlay", "portrait", "logo", "house", "pdf", "postcard", "recipe"]);
    // yhl
    print_status("yhl", base, ["overlay", "portrait", "logo", "house", "pdf", "postcard", "recipe"]);
    //rlp
    print_status("rlp", base, ["overlay", "portrait", "logo", "house", "pdf"]);
    console.log("");
}


//start
main();

